wicket-core/src/test/java/org/apache/wicket/resource/FileSystemResourceReferenceTestFileTypeDetector.java

```diff
@@ -0,0 +1,38 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.wicket.resource;
+
+import java.io.IOException;
+import java.nio.file.Path;
+import java.nio.file.spi.FileTypeDetector;
+
+/**
+ * File type detector for unit tests of file system resource reference
+ * 
+ * @author Tobias Soloschenko
+ *
+ */
+public class FileSystemResourceReferenceTestFileTypeDetector extends FileTypeDetector
+{
+
+	@Override
+	public String probeContentType(Path path) throws IOException
+	{
+		return "text/plain_provided_by_detector";
```

Comment:
Because of `wicket-core/src/test/java/META-INF/services/java.nio.file.spi.FileTypeDetector` this impl is available for *all* tests in the classpath. By using 'if' we can isolate it to do its job only for this particular test.